### PR TITLE
WRR-5250: Convert DayPicker/DaySelectorDecorator to functional component

### PR DESCRIPTION
**Pull request by stanca-pop-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Converted DayPicker/DaySelectorDecorator from class to functional component.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)
`delete rest.everyDayText;
delete rest.everyWeekdayText;
delete rest.everyWeekendText;` were deleted as they were not present in rest:
image

### Links
[//]: # (Related issues, references)
WRR-5250

### Comments
Enact-DCO-1.0-Signed-off-by: Stanca Pop stanca.pop@lgepartner.com

### CODE REVIEW

File: DayPicker/DaySelectorDecorator.js [hunk 1]
```diff
@@ -203,113 +208,100 @@ function getSelectedDayString (selected, noneText = '', dayNameLength = 'long')
...
-			const {dayNameLength, locale, selected, ...rest} = this.props;
-			const state = getLocaleState(dayNameLength, locale);
-
-			const localSelected = localizeSelected(selected, state);
-			const abbreviatedDayNames = orderDays(state.abbreviatedDayNames, state);
-			const fullDayNames = orderDays(state.fullDayNames, state);
-
-			delete rest.everyDayText;
-			delete rest.everyWeekdayText;
-			delete rest.everyWeekendText;
```
Comment (daniel-stoian-lgp) [inline DayPicker/DaySelectorDecorator.js:294]: please leave a comment in the PR description about why these are not needed anymore. i see that they are not present as props

**Review by daniel-stoian-lgp — APPROVED** [review]

File: DayPicker/DaySelectorDecorator.js [hunk 2]
```diff
@@ -203,113 +208,102 @@ function getSelectedDayString (selected, noneText = '', dayNameLength = 'long')
...
-					{abbreviatedDayNames.map((children, index) => ({
-						children,
-						// "short" dayNameLength can result in the same name so adding index
-						key: `${index} ${children}`,
-						'aria-label': fullDayNames[index]
-					}))}
-				</Wrapped>
-			);
-		}
+	DaySelector.displayName = "DaySelectorDecorator";
```
Comment (MikyungKim) [inline DayPicker/DaySelectorDecorator.js:245]:
```suggestion
	DaySelector.displayName = 'DaySelectorDecorator';
```

**Comment by MikyungKim** [comment]
Could you please update your branch with the latest develop?

**Review by MikyungKim — APPROVED** [review]
LGTM

**Pull request merged**